In [3]:
import sys

In [4]:
sys.path.append(r'C:\Users\jk\Desktop\data science - infoshare\projekt_SQL\jdszr4-edc\2-projekt-python/misc')

In [5]:
from matplotlib import pyplot
import pandas as pd
import numpy as np

In [6]:
import data_preparation as prep

In [7]:
df = prep.df
df

Age  Attrition  BusinessTravel  DailyRate  Department  DistanceFromHome  \
0      41          1               1       1102           2                 1   
1      49          0               2        279           1                 8   
2      37          1               1       1373           1                 2   
3      33          0               2       1392           1                 3   
4      27          0               1        591           1                 2   
...   ...        ...             ...        ...         ...               ...   
1465   36          0               2        884           1                23   
1466   39          0               1        613           1                 6   
1467   27          0               1        155           1                 4   
1468   49          0               2       1023           2                 2   
1469   34          0               1        628           1                 8   

      Education  EducationField  EnvironmentSatisfaction  Gender  ...  \
0             2               1                        2       0  ...   
1             1               1                        3       1  ...   
2             2               5                        4       1  ...   
3             4               1                        4       0  ...   
4             1               3                        1       1  ...   
...         ...             ...                      ...     ...  ...   
1465          2               3                        3       1  ...   
1466          1               3                        4       1  ...   
1467          3               1                        2       1  ...   
1468          3               3                        4       1  ...   
1469          3               3                        2       1  ...   

      PerformanceRating  RelationshipSatisfaction  StockOptionLevel  \
0                     3                         1                 0   
1                     4                         4                 1   
2                     3                         2                 0   
3                     3                         3                 0   
4                     3                         4                 1   
...                 ...                       ...               ...   
1465                  3                         3                 1   
1466                  3                         1                 1   
1467                  4                         2                 1   
1468                  3                         4                 0   
1469                  3                         1                 0   

      TotalWorkingYears  TrainingTimesLastYear  WorkLifeBalance  \
0                     8                      0                1   
1                    10                      3                3   
2                     7                      3                3   
3                     8                      3                3   
4                     6                      3                3   
...                 ...                    ...              ...   
1465                 17                      3                3   
1466                  9                      5                3   
1467                  6                      0                3   
1468                 17                      3                2   
1469                  6                      3                4   

      YearsAtCompany  YearsInCurrentRole  YearsSinceLastPromotion  \
0                  6                   4                        0   
1                 10                   7                        1   
2                  0                   0                        0   
3                  8                   7                        3   
4                  2                   2                        2   
...              ...                 ...                      ...   
1465           

Opis Danych:

Education
1 = 'Below College'
2 = 'College'
3 = 'Bachelor'
4 = 'Master'
5 = 'Doctor'

EducationField
'Human Resources': 0, 'Life Sciences': 1, 'Marketing': 2, 'Medical': 3, 'Technical Degree': 4, 'Other': 5

In [8]:
def corr_color(val):
    if abs(val) == 1.0:
        color = 'grey'
    elif abs(val) > 0.8:
        color = 'red'
    elif abs(val) > 0.5:
        color = 'orange'
    elif abs(val) > 0.3:
        color = 'yellow'
    else:
        color = 'blue'
    return 'color: %s' % color

# Analiza satysfakcji z pracy ze względu na zdobyty poziom edukacji - Education

In [9]:
df_Education = df[['Education', 'EnvironmentSatisfaction', 'JobSatisfaction', 'RelationshipSatisfaction']]
df_Education.sort_values(by=['Education'])

Education  EnvironmentSatisfaction  JobSatisfaction  \
734           1                        2                1   
1017          1                        2                1   
1007          1                        3                4   
618           1                        1                1   
998           1                        1                4   
...         ...                      ...              ...   
56            5                        2                1   
1028          5                        2                3   
1312          5                        4                1   
151           5                        3                2   
1138          5                        2                3   

      RelationshipSatisfaction  
734                          1  
1017                         2  
1007                         1  
618                          3  
998                          4  
...                        ...  
56                           4  
1028                         1  
1312                         3  
151                          1  
1138                         3  

[1470 rows x 4 columns]

In [10]:
df.describe()['Education']

count    1470.000000
mean        2.912925
std         1.024165
min         1.000000
25%         2.000000
50%         3.000000
75%         4.000000
max         5.000000
Name: Education, dtype: float64

In [11]:
x1 = df[(df.Education == 1)]
x1_enivronment = x1['EnvironmentSatisfaction'].mean()
x1_job = x1['JobSatisfaction'].mean()
x1_relationship = x1['RelationshipSatisfaction'].mean()
x1_overall = (x1_enivronment + x1_job +x1_relationship)/3

print(x1_enivronment,x1_job, x1_relationship, x1_overall)

2.7470588235294118 2.8 2.723529411764706 2.7568627450980387


In [12]:
x2 = df[(df.Education == 2)]
x2_enivronment = x2['EnvironmentSatisfaction'].mean()
x2_job = x2['JobSatisfaction'].mean()
x2_relationship = x2['RelationshipSatisfaction'].mean()
x2_overall = (x2_enivronment + x2_job +x2_relationship)/3

print(x2_enivronment,x2_job, x2_relationship, x2_overall)

2.7163120567375887 2.769503546099291 2.769503546099291 2.75177304964539


In [14]:
x3 = df[(df.Education == 3)]
x3_enivronment = x3['EnvironmentSatisfaction'].mean()
x3_job = x3['JobSatisfaction'].mean()
x3_relationship = x3['RelationshipSatisfaction'].mean()
x3_overall = (x3_enivronment + x3_job +x3_relationship)/3

print(x3_enivronment,x3_job, x3_relationship, x3_overall)

2.7762237762237763 2.652097902097902 2.664335664335664 2.697552447552448


In [15]:
x4 = df[(df.Education == 4)]
x4_enivronment = x4['EnvironmentSatisfaction'].mean()
x4_job = x4['JobSatisfaction'].mean()
x4_relationship = x4['RelationshipSatisfaction'].mean()
x4_overall = (x4_enivronment + x4_job +x4_relationship)/3

print(x4_enivronment,x4_job, x4_relationship, x4_overall)

2.6457286432160805 2.78643216080402 2.7462311557788945 2.726130653266331


In [16]:
x5 = df[(df.Education == 5)]
x5_enivronment = x5['EnvironmentSatisfaction'].mean()
x5_job = x5['JobSatisfaction'].mean()
x5_relationship = x5['RelationshipSatisfaction'].mean()
x5_overall = (x5_enivronment + x5_job +x5_relationship)/3

print(x5_enivronment,x5_job, x5_relationship, x5_overall)

2.6458333333333335 2.6666666666666665 2.625 2.6458333333333335


|_POZIOM EDUKACJI_ ||_ŚREDNIA SATYSFAKCJA ZE ŚRODOWISKA PRACY_ ||_ŚREDNIA SATYSFAKCJA Z PRACY_||_ŚREDNIA SATYSFAKCJA Z RELACJI W PRACY_||_ŚREDNI CAŁKOWITY POZIOM SATYSFAKCJI|
|-----||-----||-----||-----||-----|
| poniżej college'u ||2.75||2.8||2.72||2.76|
|ukończony college||2.72||2.77||2.77||2.75|
|licencjat||2.78||2.65||2.66||2.70|
|magister||2.65||2.79||2.75||2.73|
|doktor||2.65||2.67||2.62||2.65|

In [24]:
x1_corr = x1.corr()
x1_corr = x1_corr.style.applymap(corr_color)
x1_corr

Wśród osób, które nie ukończyły college'u nie zoabserwowano korealcji pomiędzy poziomem edukacji, a którymkolwiek z badanych czynników satysfakcji.

Brak doatkowych korealcji.

In [22]:
x2_corr = x2.corr()
x2_corr = x2_corr.style.applymap(corr_color)
x2_corr

In [25]:
x3_corr = x3.corr()
x3_corr = x3_corr.style.applymap(corr_color)
x3_corr

In [26]:
x4_corr = x4.corr()
x4_corr = x4_corr.style.applymap(corr_color)
x4_corr

In [27]:
x5_corr = x5.corr()
x5_corr = x5_corr.style.applymap(corr_color)
x5_corr